### DISCUSSION TITLE
Entering & Existing Transitions Help

### DISCUSSION THREAD

**Discussion by ShaneKeney** [software-mansion/react-native-reanimated#5857] [Q&A] [answered]
I have a pretty basic component which is just a banner that shows at the top of a form.  I have configured some entering and existing transitions that work as needed.  The look good for my use case so no issue there.

However, I am stuck on how to make the sibling components not collapse as the banner is "exiting".  

My parent component is something like this:

Form.tsx
```
<View>
  <FeedbackBanner ... showBanner={showBanner} />
  
  ... other components
</View>
```

When I trigger the banner to hide by setting `showBanner` to `false`, the `exiting` transition below runs but the layout immediately shifts and the animation for existing is now behind the "other components".  What is the best way to ensure the entering / existing transitions are not hidden by the other components during a layout shift etc.?  Just starting to learn using `react-native-reanimated` and can't seem to figure out the best way in this use case.

`FeedbackBanner.tsx`
```
import { Pressable, View } from 'react-native';
import Typography from '../Typography';
import SVG from '@/assets/svg';
import Animated, {
  Easing,
  LinearTransition,
  SlideInLeft,
  SlideOutRight
} from 'react-native-reanimated';

type Props = {
  title: string;
  description: string;
  onClosePress?: () => void;
  showBanner?: boolean;
};

const FeedbackBanner = ({ title, description, showBanner, onClosePress }: Props) => {
  if (!showBanner) return null;
  return (
    <Animated.View
      className="flex flex-row overflow-hidden rounded border border-Feedback-Error bg-Neutral-White dark:bg-Neutral-White"
      key={'feedback-banner'}
      exiting={SlideOutRight.duration(500).easing(Easing.ease)}
      entering={SlideInLeft.duration(500).easing(Easing.ease)}
    >
      <View className="h-[100%] w-[8px] bg-Feedback-Error dark:bg-Feedback-Error"></View>
      <View className="flex-1 flex-row justify-between gap-1 px-4 py-3">
        <View className="flex flex-col">
          <Typography font="inter-semiBold">{title}</Typography>
          <Typography className="!color-Neutral-Grey-80 dark:!color-Neutral-Grey-80">
            {description}
          </Typography>
        </View>

        <Pressable onPress={onClosePress}>
          <View>
            <SVG.Close
              className="fill-Neutral-Grey-90 dark:fill-Neutral-Grey-90"
              height={24}
              width={24}
            />
          </View>
        </Pressable>
      </View>
    </Animated.View>
  );
};

export default FeedbackBanner;
```

**Comment by MatiPl01** [comment, marked as answer]
Hey!

[Here](https://github.com/software-mansion/react-native-reanimated/discussions/2918#discussioncomment-9935297) is the example solution for this problem. You would have to use the `layout` property for all siblings of the component which is animated with `entering`/`exiting` transition.

Below you can see the adapted code to match your implementation and the result:

```tsx
import React, { useState } from 'react';
import Animated, {
  LinearTransition,
  SlideInLeft,
  SlideOutRight,
  Easing,
} from 'react-native-reanimated';
import { Button, StyleSheet, Text } from 'react-native';

export default function EmptyExample() {
  const [show, setShow] = useState(false);

  const duration = 500;

  return (
    <>
      {show && (
        <Animated.View
          layout={LinearTransition}
          exiting={SlideOutRight.duration(duration).easing(Easing.ease)}
          entering={SlideInLeft.duration(duration).easing(Easing.ease)}
          style={styles.banner}>
          <Text style={styles.bannerText}>Banner</Text>
        </Animated.View>
      )}
      <Animated.View layout={LinearTransition.delay(show ? 0 : duration)}>
        <Button title="Toggle" onPress={() => setShow((prev) => !prev)} />
      </Animated.View>
    </>
  );
}

const styles = StyleSheet.create({
  banner: {
    backgroundColor: 'red',
    justifyContent: 'center',
    alignItems: 'center',
    height: 100,
  },
  bannerText: {
    color: 'white',
    fontSize: 20,
    fontWeight: 'bold',
  },
});
```

https://github.com/software-mansion/react-native-reanimated/assets/52978053/e401cf29-d9d3-4c22-9dae-f2fa505eabf2

**Reply by MatiPl01** [reply]
@ShaneKeney I will mark this comment as answer because it provides the solution and there is currently no other way to achieve the desired behavior with layout animations.

If you don't want to use layout animations, you can just measure your banner height (with `onLayout` or [measure](https://docs.swmansion.com/react-native-reanimated/docs/advanced/measure) function) and use `useAnimatedStyle` hook to animate the height of the banner parent to create a smooth transition of all content below.
